stream-product/product-core/src/test/java/com/backbase/stream/product/service/ArrangementServiceTest.java

```diff
@@ -0,0 +1,482 @@
+        }
+
+        verify(arrangementsApi).putArrangements(any());
+    }
+
+    @Test
+    void upsertBatchArrangements() {
+        AccountArrangementItemPost request = buildAccountArrangementItemPost();
+
+        AccountBatchResponseItemExtended accountBatchResponseItemExtended = new AccountBatchResponseItemExtended()
+                .arrangementId("arr_id")
+                .resourceId("resource_id")
+                .status(BatchResponseStatusCode.HTTP_STATUS_OK);
+
+        when(arrangementsApi.postBatchUpsertArrangements(any()))
+                .thenReturn(Flux.just(accountBatchResponseItemExtended));
+
+        AccountBatchResponseItemExtended response = arrangementService.upsertBatchArrangements(List.of(request)).blockFirst();
+        Assertions.assertNotNull(response);
+        Assertions.assertEquals(accountBatchResponseItemExtended.getArrangementId(), response.getArrangementId());
+
+        verify(arrangementsApi).postBatchUpsertArrangements(any());
+    }
+
+    @Test
+    void upsertBatchArrangements_Batch_Error() {
+        AccountArrangementItemPost request = buildAccountArrangementItemPost();
+
+        AccountBatchResponseItemExtended accountBatchResponseItemExtended = new AccountBatchResponseItemExtended()
+                .arrangementId("arr_id")
+                .resourceId("resource_id")
+                .status(BatchResponseStatusCode.HTTP_STATUS_BAD_REQUEST)
+                .addErrorsItem(new ErrorItem().message("Some error"))
+                .addErrorsItem(new ErrorItem().message("Some other error"));
+
+        when(arrangementsApi.postBatchUpsertArrangements(any()))
+                .thenReturn(Flux.just(accountBatchResponseItemExtended));
+
+        try {
+            arrangementService.upsertBatchArrangements(List.of(request)).blockFirst();
+        } catch (IllegalStateException e) {
+            String errorMessage = e.getMessage();
+            Assertions.assertTrue(errorMessage.startsWith("Batch arrangement update failed: 'resource_id'"));
+            Assertions.assertTrue(errorMessage.contains("message: Some error"));
+            Assertions.assertTrue(errorMessage.contains("message: Some other error"));
+        }
+
+        verify(arrangementsApi).postBatchUpsertArrangements(any());
+    }
+
+    @Test
+    void upsertBatchArrangements_Failure() {
+        AccountArrangementItemPost request = buildAccountArrangementItemPost();
+
+        WebClientResponseException webClientResponseException = buildWebClientResponseException(HttpStatus.BAD_REQUEST, "Bad Request for upsert arrangement");
+        when(arrangementsApi.postBatchUpsertArrangements(any()))
+                .thenReturn(Flux.error(webClientResponseException));
+
+        try {
+            arrangementService.upsertBatchArrangements(List.of(request)).blockFirst();
+            Assertions.fail("should have thrown an exception");
+        } catch (ArrangementUpdateException e) {
```

Comment:
There is a better way in the junit5 to test exceptions cases, If possible can we do something like this

```
ArrangementUpdateException thrown = assertThrows(
            ArrangementUpdateException.class,
            () -> arrangementService.upsertBatchArrangements(List.of(request)));

        assertTrue(thrown.getMessage().contains("Batch arrangement update failed:"));
        ```

Reply:
Sure, we can do that... let me go ahead and update it